go/vt/topo/stats_conn.go

```diff
@@ -46,21 +53,29 @@ type StatsConn struct {
 	cell     string
 	conn     Conn
 	readOnly bool
+	readSem  *semaphore.Weighted
 }
 
 // NewStatsConn returns a StatsConn
-func NewStatsConn(cell string, conn Conn) *StatsConn {
+func NewStatsConn(cell string, conn Conn, readSem *semaphore.Weighted) *StatsConn {
 	return &StatsConn{
 		cell:     cell,
 		conn:     conn,
 		readOnly: false,
+		readSem:  readSem,
```

Comment:
Nit, but is there any reason to pass the semaphore in rather than passing in the concurrency int64 value and creating the semaphore here? It would lower the scope of the semaphore and prevent it from potentially escaping to the heap.

Reply:
@mattlord the reason for this is [the global cell has 2 x `*StatsConn` connections](https://github.com/vitessio/vitess/blob/7ae2b141d3633e8e4a04474eb723f51b0c633b81/go/vt/topo/server.go#L216-L232) and I wanted to ensure reads on both connections respect a single limit

It's possible all reads occur on the "read only" connection but nothing will stop something from reading from the main one

Reply:
Ah, OK. I missed that the semaphore was shared. Then it makes sense. Thanks!